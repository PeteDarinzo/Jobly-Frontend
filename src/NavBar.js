import React from "react";
import { NavLink } from "react-router-dom";
import "./Navbar.css"
import { Navbar, Nav, NavItem } from "reactstrap";

const NavBar = ({ loggedIn }) => {

  return (
    <div>
      <Navbar className="py-3">
        <NavLink exact to="/">
          Jobly
        </NavLink>
        {loggedIn && (
          <Nav>
            <NavItem>
              <NavLink to="/companies">
                Companies
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/jobs">
                Jobs
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/profile">
                Profile
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/logout">
                Log Out
              </NavLink>
            </NavItem>
          </Nav>
        )}
        {!loggedIn && (<Nav>
          <NavItem>
            <NavLink to="/login">
              Login
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/signup">
              Signup
            </NavLink>
          </NavItem>
        </Nav>)}
      </Navbar>
    </div >

  );
}

export default NavBar;