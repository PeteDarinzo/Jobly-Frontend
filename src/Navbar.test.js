import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";

it("renders a navbar without crashing", () => {
  render(
    <MemoryRouter>
      <NavBar loggedIn="test-token" logout={() => { }} />
    </MemoryRouter>
  )
});

it("matches the navbar snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <NavBar loggedIn="test-token" logout={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches the navbar snapshot when logged out", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <NavBar loggedIn="" logout={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});