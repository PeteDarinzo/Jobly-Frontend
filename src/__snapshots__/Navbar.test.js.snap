// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the navbar snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="py-3 navbar"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="Nav-logo active"
          href="/"
        >
          Jobly 
          <i
            class="fas fa-suitcase"
          />
        </a>
        <ul
          class="nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/login"
            >
              Log Out
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches the navbar snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="py-3 navbar"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="Nav-logo active"
          href="/"
        >
          Jobly 
          <i
            class="fas fa-suitcase"
          />
        </a>
        <ul
          class="nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/login"
            >
              Login
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/signup"
            >
              Signup
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
